iree/hal/cuda/graph_command_buffer.c

```diff
@@ -266,6 +336,24 @@ static uint32_t iree_hal_cuda_splat_pattern(const void* pattern,
   }
 }
 
+static size_t iree_cuda_get_dep_nodes(
+    iree_hal_cuda_graph_command_buffer_t* command_buffer,
+    CUgraphNode (*dep)[CUDA_MAX_LEAF_NODES]) {
+  if (command_buffer->last_sync_node == NULL) return 0;
+  (*dep)[0] = command_buffer->last_sync_node;
+  return 1;
```

Comment:
since this always returns 0 or 1 is there a reason to deal with the array? (it'd make the code easier to follow and keep the stack size smaller without it)